src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1696,25 +1721,69 @@ private static BoundExpression RemoveImplicitConversions(BoundExpression expr)
             return expr;
         }
 
-        // See Binder.GenerateConversionForAssignment for corresponding approach in initial binding.
-        private Result GenerateConversionForAssignment(BoundExpression value, TypeSymbol targetType, Result valueResult)
+        // See Binder.GenerateConversionForAssignment for initial binding.
+        private Result GenerateConversionForAssignment(BoundExpression sourceExpression, TypeSymbolWithAnnotations sourceType, TypeSymbol destinationType)
         {
-            if (targetType is null)
+            if (destinationType is null)
             {
                 return Result.Unset;
             }
-
-            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-            TypeSymbol sourceType = valueResult.Type?.TypeSymbol;
-            var conversion = (sourceType is null) ?
-                _conversions.ClassifyImplicitConversionFromExpression(value, targetType, ref useSiteDiagnostics) :
-                _conversions.ClassifyImplicitConversionFromType(sourceType, targetType, ref useSiteDiagnostics);
+            var type = sourceType?.TypeSymbol;
+            var conversion = GenerateConversion(_conversions, sourceExpression, type, destinationType);
             if (!conversion.Exists)
             {
-                ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullabilityMismatchInAssignment, value.Syntax, GetTypeAsDiagnosticArgument(valueResult.Type?.TypeSymbol), targetType);
+                ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullabilityMismatchInAssignment, sourceExpression.Syntax, GetTypeAsDiagnosticArgument(type), destinationType);
             }
 
-            return InferResultNullability(value, conversion, targetType, valueResult.Type);
+            return InferResultNullability(sourceExpression, conversion, destinationType, sourceType);
+        }
+
+        // See Binder.BindNullCoalescingOperator for initial binding.
+        private bool GenerateConversionForConditionalOperator(BoundExpression sourceExpression, TypeSymbol sourceType, TypeSymbol destinationType, bool reportMismatch)
```

Comment:
>GenerateConversionForConditionalOperator [](start = 21, length = 40)

Consider using different name. We have three methods that start from GenerateConversion and all of them return different types. This is very confusing. Consider renaming all of them so that it would be clear what do they produce

Reply:
Inlined first overload and changed second overload to return `Conversion` to match the third.

---
In reply to: [175207821](https://github.com/dotnet/roslyn/pull/25358#discussion_r175207821) [](ancestors = 175207821)